src/core/Akka/Actor/FSM.cs

```diff
@@ -539,15 +540,16 @@ public State(TS stateName, TD stateData, TimeSpan? timeout = null, Reason stopRe
             /// </summary>
             internal State<TS, TD> WithNotification(bool notifies)
             {
+                // don't bother allocating even a stack type if the notifies value is identical.
+                if (Notifies == notifies)
```

Comment:
Don't return a new copy of this object if the current `Notifies`  value matches the new one (it usually does.)